components/textfield/index.css

```diff
@@ -445,42 +444,42 @@ governing permissions and limitations under the License.
 	display: inline-grid;
 	grid-template-columns: auto auto;
 	grid-template-rows: auto auto auto;
+}
 
-	&.spectrum-Textfield--quiet {
+.spectrum-Textfield.spectrum-Textfield--quiet {
+	&::after {
+		content: "";
+		position: absolute;
+		inset-block-end: calc(
+			-1 * (var(
+							--mod-textfield-focus-indicator-gap,
+							var(--spectrum-textfield-focus-indicator-gap)
+						) +
+						var(
+							--mod-textfield-focus-indicator-width,
+							var(--spectrum-textfield-focus-indicator-width)
+						))
+		);
+		inset-inline-start: 0;
+		inline-size: 100%;
+		block-size: var(
+			--mod-textfield-focus-indicator-width,
+			var(--spectrum-textfield-focus-indicator-width)
+		);
+	}
+
+	&.is-keyboardFocused,
+	&:focus-visible,
+	&:focus-within {
 		&::after {
-			content: "";
-			position: absolute;
-			bottom: calc(
-				-1 * (var(
-								--mod-textfield-focus-indicator-gap,
-								var(--spectrum-textfield-focus-indicator-gap)
-							) +
-							var(
-								--mod-textfield-focus-indicator-width,
-								var(--spectrum-textfield-focus-indicator-width)
-							))
-			);
-			left: 0;
-			inline-size: 100%;
-			block-size: var(
-				--mod-textfield-focus-indicator-width,
-				var(--spectrum-textfield-focus-indicator-width)
+			background-color: var(
+				--highcontrast-textfield-focus-indicator-color,
+				var(
+					--mod-textfield-focus-indicator-color,
+					var(--spectrum-textfield-focus-indicator-color)
+				)
```

Comment:
Can I ask a favor for this section? Would you mind removing some of the whitespace and indenting that Prettier introduced? IMO it's a little _less pretty_ now...